server/src/main/java/org/elasticsearch/common/util/BigArrays.java

```diff
@@ -261,6 +261,13 @@ public void set(long index, byte[] buf, int offset, int len) {
             assert index >= 0 && index < size();
             System.arraycopy(buf, offset << 3, array, (int) index << 3, len << 3);
         }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            int size = (int) size();
```

Comment:
Maybe use `Math.toIntExcact(sizze())` here to make this obviously correct?

Reply:
Thanks for the review! I pushed the changes via: ebe635c3902ead50040a8a5d879015231867f4e1